### ISSUE TITLE
[a11y] Focus state / keyboard nav broken for many areas

### ISSUE THREAD

**Issue by pbakaus** [ampproject/amp.dev#2587]
### Current Behavior
Tabbing through amp.dev with a keyboard is a frustrating experience, tabs into many areas that are invisible.

Please also provide:
- Page link: amp.dev

### Expected behavior/code
There should be a focus state for every element with a tab index,  and focus should not go into elements that are not visible.

### How do we reproduce the issue?
Go to amp.dev on mobile or mobile emulation, try to tab to reach the hamburger menu.

Hint, CSS like this is part of the issue:

```
.ap-o-burger-menu-label:focus {
    outline: 0;
}
```

**Comment by sebastianbenz** [comment]
Thanks @sebil for the quick fix! There are a few more areas:

1. sample overview page: https://amp-dev-staging.appspot.com/documentation/examples/ (not sure if this is going to be better after @dlemm's PR has landed.
2. sidebar: compound guides cannot be expanded via keyboard )https://amp-dev-staging.appspot.com/documentation/guides-and-tutorials/?format=websites)
3. sidebar cannot be closed via keyboard

**Comment by sebil** [comment]
Hi @sebastianbenz 

according to our status call this morning, I talked to @LongoOfficial about a new solution for the docs `sidebar` and `sidebar-toggle` for desktop.
> 3. sidebar cannot be closed via keyboard

We want to rethink the whole idea of toggling the sidebar, but I think this is a new major task and should be separated from this accessibility issue.
I've also already fixed the other two points you mentioned. They will be merged together with #2479 asap.

I'd come back to you with new solutions for the sidebar/toggle, so I'd suggest to close this issue here.

Are you fine with that?
